lib/board.rb

```diff
@@ -0,0 +1,47 @@
+class Board
+
+  attr_accessor :cells
+
+  @cells = []
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(input)
+    cells[input.to_i-1]
+  end
+
+  def update(input, player)
+    cells[input.to_i-1] = player.token
+  end
+
+  def full?
+    cells.all?{|position| position == "X" || position == "O"}
+  end
+
+  def turn_count
+    cells.count("X") + cells.count("O")
```

Comment:
This can also be accomplished with one call to `#count`. See below:
```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```